ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift

```diff
@@ -90,15 +120,19 @@ class KeymanWebViewController: UIViewController {
   }
 
   override func viewWillAppear(_ animated: Bool) {
-    view.frame = CGRect(origin: .zero, size: keyboardSize)
     super.viewWillAppear(animated)
   }
+  
+  // Very useful for immediately adjusting the WebView's properties upon loading.
+  override func viewDidAppear(_ animated: Bool) {
+    fixLayout()
+  }
 }
 
 // MARK: - JavaScript functions
 extension KeymanWebViewController {
   func languageMenuPosition(_ completion: @escaping (CGRect) -> Void) {
-    webView.evaluateJavaScript("langMenuPos();") { result, _ in
+    webView!.evaluateJavaScript("langMenuPos();") { result, _ in
```

Comment:
Sometimes you use `webView!` and others you use `webView?`.  Is this intentional and if so it might be good to clarify why.

Reply:
That's a Swift thing - `webView!` asserts that it's not `nil`, while `webView?` has a built-in `nil` check before continuing with the rest of the line.

Reply:
Yep -- it's just not clear to me the reason why sometimes you assert and sometimes you go ahead with a nil check for each of those JS calls.

Reply:
OK.  There are two cases I see in that file, and they do have a particular reason - they're the `setOSKWidth`/`setOSKHeight` methods, right?  Those two lines actually _can_ be reached before `webView` is initialized, so they need the `nil` check.  It's also fine if the call doesn't go through, as a state variable is also set with the size information and can be passed along to the `webView` as part of its initialization, so it's not like the values are being lost.

Without other changes, setting those two checks to use `!` instead of `?` will actually crash the app.